[wifiled] Scheduled runnable ended with exception

## Current Behavior
It does not seem to affect functionality, but when OH starts, there is an exception...

```
2020-01-17 07:11:09.780 [WARN ] [org.eclipse.smarthome.core.internal.common.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
java.lang.ArrayIndexOutOfBoundsException: 1
        at org.openhab.binding.wifiled.internal.discovery.WiFiLEDDiscoveryService.discover(WiFiLEDDiscoveryService.java:107) ~[?:?]
        at org.openhab.binding.wifiled.internal.discovery.WiFiLEDDiscoveryService.lambda$0(WiFiLEDDiscoveryService.java:64) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_232]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_232]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_232]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[?:1.8.0_232]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232
```

## Steps to Reproduce (for Bugs)
I have 2 Arilux AL-LC02 controllers configured as LD382A Things. All I do is restart OH to reproduce the issue. This has been happening for a long while.

## Your Environment

* Version used: (e.g., openHAB and add-on versions) OH 2.5.1-2
* Operating System and version (desktop or mobile, Windows 10, Raspbian Buster, ...): FedoraServer 31

Hello,

noted the same issue using OH 2.5.0 final using delivered WIFILED binding and Magic UFO-Wifi-led controller LD382A

**the autodiscovery of WIFI LED binding does not work. .(NOK)**

this was working without problems in OH 2.5.0 milestone 3

click on SCAN in PaperUI:

```
2020-01-21 21:40:15.300 [DEBUG] [al.discovery.WiFiLEDDiscoveryService] - Start WiFi LED scan
2020-01-21 21:40:15.302 [DEBUG] [al.discovery.WiFiLEDDiscoveryService] - Try to discover all WiFi LED devices
2020-01-21 21:40:15.305 [DEBUG] [al.discovery.WiFiLEDDiscoveryService] - Disover message sent: '48 46 2d 41 31 31 41 53 53 49 53 54 48 52 45 41 44'
2020-01-21 21:40:15.315 [DEBUG] [al.discovery.WiFiLEDDiscoveryService] - Disover response received: 'HF-A11ASSISTHREAD' [48 46 2d 41 31 31 41 53 53 49 53 54 48 52 45 41 44 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00]
2020-01-21 21:40:15.317 [ERROR] [nternal.DiscoveryServiceRegistryImpl] - Cannot trigger scan for thing types '[wifiled:wifiled]' on 'WiFiLEDDiscoveryService'!
java.lang.ArrayIndexOutOfBoundsException: 1
        at org.openhab.binding.wifiled.internal.discovery.WiFiLEDDiscoveryService.discover(WiFiLEDDiscoveryService.java:107) ~[?:?]
```

The WiFiLEDDiscoveryService expects to receive a value that consists of 3 parts: IP Address, MAC Address, and Hostname:

192.168.178.25,ACCF23489C9A,HF-LPB100-ZJ200

log shows not getting that:

2020-01-21 21:31:26.582 [DEBUG] [al.discovery.WiFiLEDDiscoveryService] - Disover response received: 'HF-A11ASSISTHREAD' 



startup exception also always:

```
2020-01-20 21:59:07.151 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.ArrayIndexOutOfBoundsException: 1
at org.openhab.binding.wifiled.internal.discovery.WiFiLEDDiscoveryService.discover(WiFiLEDDiscoveryService.java:107) ~[?:?]
at org.openhab.binding.wifiled.internal.discovery.WiFiLEDDiscoveryService.lambda$0(WiFiLEDDiscoveryService.java:64) ~[?:?]
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_201]
at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_201]
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_201]
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[?:1.8.0_201]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_201]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_201]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_201]
```


Please check this issue. Thanks!

--- a/bundles/org.openhab.binding.wifiled/README.md
+++ b/bundles/org.openhab.binding.wifiled/README.md
@@ -46,19 +46,17 @@ While the CLASSIC driver lets you choose and run device internal programs (e.g. 
 all normal operations (e.g. turn on or off, switch color, etc.) are performed immediately and
 without any fading effect.
 
-If you prefer to switch colors smoothly and to turn your light on and off by slightly increasing/decreasing the brightness,
-you should try the FADING driver.  If selected, you can also set the number of fading steps and the fading duration in the
-Thing configuration.  Each fading step will at least take 10 ms to be processed. This limit comes from the speed of the LED
-controller and your network speed. Thus a color fading with a configured fading duration of 0s might still take some time;
-count on more than 1 second for 100 steps.  If the FADING driver is chosen, the program and the programSpeed channels will
-not have any effect.
+If you prefer to switch colors smoothly and to turn your light on and off by slightly increasing/decreasing the brightness, you should try the FADING driver.
+If selected, you can also set the number of fading steps and the fading duration in the Thing configuration.
+Each fading step will at least take 10 ms to be processed. This limit comes from the speed of the LED controller and your network speed.
+Thus a color fading with a configured fading duration of 0s might still take some time; count on more than 1 second for 100 steps.
+If the FADING driver is chosen, the program and the programSpeed channels will not have any effect.
 
-The polling period is a parameter only used by the CLASSIC driver and specifies the time in seconds after the LED state is
-refreshed in openHAB.
+The polling period is a parameter only used by the CLASSIC driver and specifies the time in seconds after the LED state is refreshed in openHAB.
 
 ### Device Discovery
 
-If the automatic discovery fails, you have to set the IP address and the port of your device manually.  Make sure that the
+If the automatic discovery fails, you have to set the IP address and the port of your device manually.  Make sure that the 
 device protocol matches your device type.
 
 ## Channels

--- a/bundles/org.openhab.binding.wifiled/README.md
+++ b/bundles/org.openhab.binding.wifiled/README.md
@@ -48,7 +48,8 @@ without any fading effect.
 
 If you prefer to switch colors smoothly and to turn your light on and off by slightly increasing/decreasing the brightness, you should try the FADING driver.
 If selected, you can also set the number of fading steps and the fading duration in the Thing configuration.
-Each fading step will at least take 10 ms to be processed. This limit comes from the speed of the LED controller and your network speed.
+Each fading step will at least take 10 ms to be processed.
+This limit comes from the speed of the LED controller and your network speed.
 Thus a color fading with a configured fading duration of 0s might still take some time; count on more than 1 second for 100 steps.
 If the FADING driver is chosen, the program and the programSpeed channels will not have any effect.
 

--- a/bundles/org.openhab.binding.wifiled/README.md
+++ b/bundles/org.openhab.binding.wifiled/README.md
@@ -29,7 +29,7 @@ No binding configuration required.
 
 ## Thing Configuration
 
-The thing can be configured through the Paper UI.
+The Thing can be configured through the Paper UI.
 Use the configuration if you have devices of type LD382 or LD686, want to enable color fading,
 or if the device discovery does not find your LED controller automatically.
 
@@ -42,24 +42,24 @@ You can choose between two drivers with different functionality:
 | CLASSIC |                       |         ✓         |        ✓        |
 | FADING  |            ✓          |                   |                 |
 
-While the CLASSIC driver let you choose and run device internal programs (e.g. alternating blue),
-all normal operations (turn on or off, switch color, set brightness, ...) are performed immediately
-and without any fading effect.
+While the CLASSIC driver lets you choose and run device internal programs (e.g. alternating blue),
+all normal operations (e.g. turn on or off, switch color, etc.) are performed immediately and
+without any fading effect.
 
-If you prefer to switch colors smoothly and to turn your light on and off by slightly rising/decreasing the brightness
-you should try the FADING driver.
-If selected you can also set the number of fading steps and the fading duration in the thing configuration.
-Note that each fading step will at least take 10 ms for being processed.
-This natural limit is given by the speed of the LED controller and your network speed.
-Thus, a color fading with a configured fading duration of 0s might still take some time (count with more than 1 second for 100 steps).
-IF the "FADING" driver is chosen the program channel and the programSpeed channel will not have any effect.
+If you prefer to switch colors smoothly and to turn your light on and off by slightly increasing/decreasing the brightness,
+you should try the FADING driver.  If selected, you can also set the number of fading steps and the fading duration in the
+Thing configuration.  Each fading step will at least take 10 ms to be processed. This limit comes from the speed of the LED
+controller and your network speed. Thus a color fading with a configured fading duration of 0s might still take some time;
+count on more than 1 second for 100 steps.  If the FADING driver is chosen, the program and the programSpeed channels will
+not have any effect.
 
-The polling period is a parameter only used by the CLASSIC driver and specifies a the time in seconds after the LED state is refreshed in openHAB.
+The polling period is a parameter only used by the CLASSIC driver and specifies the time in seconds after the LED state is
+refreshed in openHAB.
 
 ### Device Discovery
 
-If the automatic discovery fails you have to set the IP address and the port of your device manually.
-Moreover, make sure that the device protocol matches you device type.
+If the automatic discovery fails, you have to set the IP address and the port of your device manually.  Make sure that the
+device protocol matches your device type.
 
 ## Channels
 
@@ -75,7 +75,7 @@ Moreover, make sure that the device protocol matches you device type.
 ## Example
 
 Usually, there is no need to define your WiFi LED controllers via configuration files.
-However, if you like to do it, here is an example.
+However, here is an example.
 
 wifiled.things:
 

--- a/bundles/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/internal/discovery/WiFiLEDDiscoveryService.java
+++ b/bundles/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/internal/discovery/WiFiLEDDiscoveryService.java
@@ -104,7 +104,7 @@ public class WiFiLEDDiscoveryService extends AbstractDiscoveryService {
                 // ^-IP..........,^-MAC.......,^-HOSTNAME.....
 
                 String[] ss = s.split(",");
-                if( ss.length < 3 ) {
+                if (ss.length < 3) {
                     logger.debug("Ignoring unparseable discovery response: '{}'", s);
                     continue;
                 }

--- a/bundles/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/internal/discovery/WiFiLEDDiscoveryService.java
+++ b/bundles/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/internal/discovery/WiFiLEDDiscoveryService.java
@@ -46,7 +46,7 @@ import org.slf4j.LoggerFactory;
 public class WiFiLEDDiscoveryService extends AbstractDiscoveryService {
 
     private static final int DEFAULT_BROADCAST_PORT = 48899;
-
+    private static final String DISCOVER_MESSAGE = "HF-A11ASSISTHREAD";
     private Logger logger = LoggerFactory.getLogger(WiFiLEDDiscoveryService.class);
 
     public WiFiLEDDiscoveryService() {
@@ -80,10 +80,10 @@ public class WiFiLEDDiscoveryService extends AbstractDiscoveryService {
             InetAddress inetAddress = InetAddress.getByName("255.255.255.255");
 
             // send discover
-            byte[] discover = "HF-A11ASSISTHREAD".getBytes();
+            byte[] discover = DISCOVER_MESSAGE.getBytes();
             DatagramPacket packet = new DatagramPacket(discover, discover.length, inetAddress, DEFAULT_BROADCAST_PORT);
             socket.send(packet);
-            logger.debug("Disover message sent: '{}'", ClassicWiFiLEDDriver.bytesToHex(discover));
+            logger.debug("Discover message sent: '{}'", DISCOVER_MESSAGE);
 
             // wait for responses
             while (true) {
@@ -92,17 +92,23 @@ public class WiFiLEDDiscoveryService extends AbstractDiscoveryService {
                 try {
                     socket.receive(packet);
                 } catch (SocketTimeoutException e) {
-                    break; // leave the endless loop
+                    logger.trace("Timeout exceeded. Discovery process ended.");
+                    break;
                 }
 
                 byte[] data = packet.getData();
                 String s = bytesToString(data);
-                logger.debug("Disover response received: '{}' [{}] ", s, ClassicWiFiLEDDriver.bytesToHex(data));
+                logger.debug("Discovery response received: '{}' [{}] ", s, ClassicWiFiLEDDriver.bytesToHex(data));
 
                 // 192.168.178.25,ACCF23489C9A,HF-LPB100-ZJ200
                 // ^-IP..........,^-MAC.......,^-HOSTNAME.....
 
                 String[] ss = s.split(",");
+                if( ss.length < 3 ) {
+                    logger.debug("Ignoring unparseable discovery response: '{}'", s);
+                    continue;
+                }
+
                 String ip = ss[0];
                 String mac = ss[1];
                 String name = ss[2];
@@ -118,7 +124,7 @@ public class WiFiLEDDiscoveryService extends AbstractDiscoveryService {
                 logger.debug("Thing discovered '{}'", result);
             }
         } catch (IOException e) {
-            logger.debug("No WiFi LED device found. Diagnostic: {}", e.getMessage());
+            logger.debug("Device discovery encountered an I/O Exception: {}", e.getMessage(), e);
         }
     }
 

